src/mlpack/core/cv/metrics/precision.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file precision.hpp
+ * @author Kirill Mishchenko
+ *
+ * The precision metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_PRECISION_HPP
+#define MLPACK_CORE_CV_METRICS_PRECISION_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/cv/metrics/average_strategy.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * Precision is a metric of performance for classification algorithms that for
+ * binary classification is equal to tp / (tp + fp), where tp and fp are the
+ * numbers of true positives and false positives respectively. For multiclass
+ * classification the precision metric can be used with the following strategies
+ * for averaging.
+ * 1. Micro. If there are N + 1 classes in total, the result is equal to
+ * (tp0 + tp1 + ... + tpN) / (tp0 + tp1 + ... + tpN + fp0 + fp1 + ... fpN),
+ * where tpI and fpI are the numbers of true positives and false positives
+ * respectively for the class (label) I.
```

Comment:
You can use inline LaTeX here with Doxygen, it would probably look nicer in the resulting generated documentation:

```
an inline formula is @f$ x + y @f$, and a standalone one is

@f[
(tp_0 + tp_1 + \ldots + tp_N) / (\ldots)
@f]
```

I think this comment would apply a few other places too.